cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -45,13 +46,16 @@
  * Git hosting services and provides version control operations.
  */
 public class RepositoryManager extends AbstractIdleService {
-  private final SecureStore secureStore;
   private final SourceControlConfig sourceControlConfig;
   private Git git;
+  private final CredentialsProvider credentialsProvider;
 
   public RepositoryManager(SecureStore secureStore, SourceControlConfig sourceControlConfig) {
-    this.secureStore = secureStore;
     this.sourceControlConfig = sourceControlConfig;
+    this.credentialsProvider = sourceControlConfig.getAuthenticationStrategy()
+      .getCredentialsProvider(secureStore,
+                              sourceControlConfig.getRepositoryConfig(),
```

Comment:
We're getting something from the config that requires passing fields from the config. This is a sign that the class design is not right.

It would be better to keep logic out of any of the config classes, removing the getAuthenticationStrategy() method from it. Instead, move it to a AuthenticationStrategyProvider class that takes an AuthConfig and returns an AuthenticationStrategy. Something like:

```
public class AuthenticationStrategyProvider {
  private final SecureStore secureStore;
  private final NamespaceId namespaceId;

  AuthenticationStrategy get(AuthConfig config) {
    // logic from getAuthenticationStrategy()
  } 
}
```

Also, the current logic checks that the provider is github and the auth type is PAT. It should not be based on a combination of two variables. Why is the provider important at all, isn't it all git?

Reply:
Yeah I think we could remove the provider field and not checking it @arjan-bal

Added a ticket for tracking: https://cdap.atlassian.net/browse/CDAP-20360

Reply:
Done